components/snowflake/sources/new-row/new-row.js

```diff
@@ -0,0 +1,54 @@
+const common = require("../common-table-scan");
+
+module.exports = {
+  ...common,
+  key: "snowflake-new-row",
+  name: "New Row",
+  description: "Emit an event when a new row is added to a table",
+  methods: {
+    ...common.methods,
+    generateMeta(data) {
+      const {
+        row: {
+          [this.uniqueKey]: id,
+        },
+        timestamp: ts,
+      } = data;
+      const summary = `New row: ${id}`;
+      return {
+        id,
+        summary,
+        ts,
+      };
+    },
+    generateMetaForCollection(data) {
+      const {
+        lastResultId: id,
+        rowCount,
+        timestamp: ts,
+      } = data;
+      const summary = `${rowCount} new rows`;
+      return {
+        id,
+        summary,
+        ts,
+      };
+    },
+    async processEvent(lastResultId, event) {
+      const { timestamp } = event;
+      const sqlText = `
+        SELECT *
+        FROM ${this.tableName}
```

Comment:
this is quite spooky without prepared statements... is there no lightweight node lib that does it? (if snowflake sdk is causing issues with this)

Reply:
It's very tricky to implement on the client side, especially when dealing with custom types for example.
I don't like this either TBH, as it doesn't handle all data types (`VARCHAR` for example will break this).
I will spend some more time looking into possible alternatives.

Reply:
OK, I pushed a change that makes me feel more comfortable since it uses prepared statements.
To make it work I additionally do this:
- Use the [`IDENTIFIER`](https://docs.snowflake.com/en/sql-reference/identifier-literal.html) function to refer to the table using a prepared statement
- Validate a column name against the existing columns in a table (requires an extra API call in each execution)